packages/backend/src/models/DashboardModel/DashboardModel.ts

```diff
@@ -130,90 +138,117 @@ export class DashboardModel {
-                uuid: tab.uuid,
-                dashboard_id: dashboardId,
-                order: tab.order,
-            });
-        });
-
-        await Promise.all(tabsPromises);
+        if (version.tabs.length > 0) {
+            await trx(DashboardTabsTableName).insert(
+                version.tabs.map(async (tab) => ({
+                    dashboard_version_id: versionId.dashboard_version_id,
+                    name: tab.name,
+                    uuid: tab.uuid,
+                    dashboard_id: dashboardId,
+                    order: tab.order,
+                })),
+            );
+        }
 
-        const tilePromises = version.tiles.map(async (tile) => {
-            const { uuid: dashboardTileId, type, w, h, x, y, tabUuid } = tile;
+        const tilesWithUuids: Array<
+            | (CreateDashboardChartTile & { uuid: string })
+            | (CreateDashboardMarkdownTile & { uuid: string })
+            | (CreateDashboardLoomTile & { uuid: string })
+        > = version.tiles.map((tile) => ({
+            ...tile,
+            uuid: tile.uuid || uuidv4(),
+        }));
 
-            const [insertedTile] = await trx(DashboardTilesTableName)
-                .insert({
+        if (tilesWithUuids.length > 0) {
+            await trx(DashboardTilesTableName).insert(
+                tilesWithUuids.map(({ uuid, type, w, h, x, y, tabUuid }) => ({
                     dashboard_version_id: versionId.dashboard_version_id,
-                    dashboard_tile_uuid: dashboardTileId,
+                    dashboard_tile_uuid: uuid,
                     type,
                     height: h,
                     width: w,
                     x_offset: x,
                     y_offset: y,
                     tab_uuid: tabUuid,
-                })
-                .returning('*');
+                })),
+            );
+        }
 
-            switch (tile.type) {
-                case DashboardTileTypes.SAVED_CHART: {
-                    const chartUuid = tile.properties.savedChartUuid;
-                    if (chartUuid) {
-                        const [savedChart] = await trx(SavedChartsTableName)
-                            .select(['saved_query_id'])
-                            .where('saved_query_uuid', chartUuid)
-                            .limit(1);
-                        if (!savedChart) {
-                            throw new NotFoundError('Saved chart not found');
+        const chartTiles = tilesWithUuids.filter(isDashboardChartTileType);
+        if (chartTiles.length > 0) {
+            const chartIds = await trx(SavedChartsTableName)
+                .select(['saved_query_id', 'saved_query_uuid'])
+                .whereIn(
+                    'saved_query_uuid',
+                    chartTiles.map(
+                        ({ properties }) => properties.savedChartUuid,
+                    ),
+                );
+
+            await trx(DashboardTileChartTableName).insert(
+                tilesWithUuids
```

Comment:
we could extract this big map or make a function to simplify the insert method, it should not affect performance though, just for readability.